addons/binding/org.openhab.binding.ikeatradfri/src/main/java/org/openhab/binding/ikeatradfri/handler/IkeaTradfriGatewayHandler.java

```diff
@@ -0,0 +1,339 @@
+        try {
+            URI uri = new URI("coaps://" + configuration.host + "//" + url);
+
+            CoapClient client = new CoapClient(uri);
+            client.setEndpoint(endPoint);
+            CoapHandler handler = new CoapHandler() {
+                @Override
+                public void onLoad(CoapResponse response) {
+                    logger.debug("COAP Observe: \noptions: {}\npayload: {} ", response.getOptions().toString(),
+                            response.getResponseText());
+                    if (response.isSuccess()) {
+                        try {
+                            listener.onDataUpdate(parser.parse(response.getResponseText()));
+                        } catch (JsonParseException e) {
+                            logger.warn("Observed value not json: {}, {}", response.getResponseText(), e.getMessage());
+                        }
+                    } else {
+                        logger.debug("COAP Observe Error: {} for {}", response.getCode().toString(), url);
+                    }
+                    updateStatus(ThingStatus.ONLINE);
+                }
+
+                @Override
+                public void onError() {
+                    logger.debug("COAP Observe error");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                }
+            };
+
+            CoapObserveRelation relation = client.observe(handler);
+            observeRelationMap.put(thingUID, relation);
+
+        } catch (URISyntaxException e) {
+            logger.warn("COAP URL error: {}", e.getMessage());
+        }
+    }
+
+    private void observeDevice(ThingUID thingUID, IkeaTradfriObserveListener listener) {
+        String url = "15001/" + thingUID.getId();
+        observe(url, thingUID, listener);
+    }
+
+    private void stopObserve(ThingUID thingUID) {
+        if (observeRelationMap.containsKey(thingUID)) {
+            CoapObserveRelation relation = observeRelationMap.get(thingUID);
+            relation.proactiveCancel();
+            observeRelationMap.remove(thingUID);
+        }
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        logger.debug("Child handler initialized: {} handler: {}", childThing.getThingTypeUID().toString(),
+                childHandler);
+        if (childHandler instanceof IkeaTradfriBulbHandler) {
+            if (isInitialized() && endPoint != null) {
+                observeDevice(childThing.getUID(), (IkeaTradfriBulbHandler) childHandler);
+            } else {
+                pendingObserve.add(childThing.getUID());
+            }
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        logger.debug("Child handler disposed: {}", childThing.getThingTypeUID().toString());
+        deviceDiscoverHelper(childThing.getUID().getId());
+    }
```

Comment:
I think so - and have added it.
Why is a discovery triggered here immediately, do you know?